galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+            # than it does to create an instantaneous PSF, so we exchange the order of the PSF and
+            # time loops so we're not recomputing atmospheres needlessly when we go from PSF1 to
+            # PSF2 and so on.  For FrozenAtmosphericScreen, there's not much difference with either
+            # loop order, so we just always exchange the orders.
+            for i in xrange(_nstep):
+                for PSF in PSFs:
+                    PSF._step()
+                self.advance()
+
+            for PSF in PSFs:
+                PSF._finalize(flux, gsparams)
+            return PSFs
+
+    @property
+    def r0_500_effective(self):
+        return sum(l.r0_500**(-5./3) for l in self if hasattr(l, 'r0_500'))**(-3./5)
+
+
+class PhaseScreenPSF(GSObject):
+    """A PSF surface brightness profile constructed by integrating over time the instantaneous PSF
+    derived from a set of phase screens and an aperture.
+
+    There are at least three ways construct a PhaseScreenPSF given a PhaseScreenList.  The following
+    two statements are equivalent:
+    > psf = screen_list.getPSF(...)
+    > psf = PhaseScreenPSF(screen_list, ...)
+
+    The third option is to use screen_list.getPSF() to obtain multiple PSFs at different field
+    angles from the same PhaseScreenList over the same simulated time interval.  Depending on the
+    details of PhaseScreenList and other arguments, this may be significantly faster than manually
+    looping over getPSF().
+    > psfs = screen_list.getPSF(..., theta_x=[...], theta_y=[...])
+
+    Note that constructing a PhaseScreenPSF advances each PhaseScreen in `screen_list`, so PSFs
+    constructed consecutively with the same arguments will generally be different.  Use
+    PhaseScreenList.reset() to reset the time to t=0.
+
+    > screen_list.reset()
+    > psf1 = screen_list.getPSF(...)
+    > psf2 = screen_list.getPSF(...)
+    > assert psf1 != psf2
+    > screen_list.reset()
+    > psf3 = screen_list.getPSF(...)
+    > assert psf1 == psf3
+
+    @param screen_list      PhaseScreenList object from which to create PSF.
+    @param lam              Wavelength in nanometers used to compute PSF.  [Default: 500]
+    @param exptime          Time in seconds overwhich to accumulate evolving instantaneous PSF.
+                            [Default: 0.0]
+    @param flux             Flux of output PSF [Default: 1.0]
+    @param theta_x          x-component of field angle at which to evaluate phase screens and
+                            resulting PSF.  [Default: 0.0*galsim.arcmin]
+    @param theta_y          y-component of field angle at which to evaluate phase screens and
+                            resulting PSF.  [Default: 0.0*galsim.arcmin]
+    @param scale_unit       Scale of output profile.  [Default: galsim.arcsec]
+    @param diam             Diameter in meters of aperture used to compute PSF from phases.
+                            [Default: 8.4]
```

Comment:
Should this have a default?  And if so, should it really be 8.4?  I think maybe make this a required parameter, since we really need to know it to do this right.